src/mito2/src/cache/file_cache.rs

```diff
@@ -100,34 +100,45 @@ impl FileCache {
         self.memory_index.insert(key, value).await;
     }
 
+    async fn get_reader(&self, file_path: &str) -> object_store::Result<Option<Reader>> {
+        if self.local_store.is_exist(file_path).await? {
+            Ok(Some(self.local_store.reader(file_path).await?))
+        } else {
+            Ok(None)
+        }
+    }
+
     /// Reads a file from the cache.
     pub(crate) async fn reader(&self, key: IndexKey) -> Option<Reader> {
-        if !self.memory_index.contains_key(&key) {
+        if self.memory_index.get(&key).await.is_none() {
```

Comment:
I belive `contains_key` is better.

Reply:
We must use get here.
https://docs.rs/moka/latest/moka/future/struct.Cache.html#method.contains_key

> Unlike the get method, this method is not considered a cache read operation, so it does not update the historic popularity estimator or reset the idle timer for the key.

Reply:
I'll address it at https://github.com/GreptimeTeam/greptimedb/pull/3100
cc @WenyXu